galsim/compound.py

```diff
@@ -902,3 +904,162 @@ def __setstate__(self, d):
 _galsim.SBFourierSqrt.__setstate__ = lambda self, state: 1
 _galsim.SBFourierSqrt.__repr__ = lambda self: \
         'galsim._galsim.SBFourierSqrt(%r, %r)'%self.__getinitargs__()
+
+
+class RandomWalk(Sum):
+    """
+
+    A class for generating a set of point sources distributed using a random
+    walk.  Uses of this profile include representing an "irregular" galaxy, or
+    adding this profile to an Exponential to represent knots of star formation.
+
+    Random walk profiles have a "shape noise" of g~0.05, but can be sheared to
+    give an additional ellipticity, for example to follow that of an associated
+    disk.
+    
+    Initialization
+    --------------
+    @param  npoints  Number of point sources to generate.
+    @param  hlr      Half light radius of the distribution of points.  This
+                     is the mean half light radius produced by an 
+                     infinite number of points.  A single instance will be
+                     noisy.
+    @param  flux     Total flux in all point sources.  Default 1
+    @param  nstep    Number of steps in random walk.  Default 40
+    @param  rng      Optional random number generator. Should be a UniformDeviate.
+    @param  gsparams GSParams for the gaussians representing each point source.
+
+    Methods
+    -------
+
+    This class inherits from galsim.Sum. Additional methods, implemented as
+    read-only properties, are provided as "getters" for the basic parameters:
+
+        .npoints
+        .half_light_radius
```

Comment:
I wonder if this should have a slightly different name than just `half_light_radius`.  Other profiles that have this return an exact value, whereas this profile will have a half-light radius different from this due to noise in the random walk.  

Maybe `input_half_light_radius`?  It's a bit verbose, but I'm concerned that people will expect `rw.half_light_radius` to give them a true value for this particular profile, rather than just an input parameter that doesn't quite match the realized value.